fix: test_push_subscription_owner_validation fails — accounts_root override prevents delete

## Bug

`test_push_subscription_owner_validation` in `tests/test_push_subscription_route.py` fails because the test sets `client.app.state.accounts_root` to a non-existent path before calling DELETE, then asserts the subscription was removed.

The DELETE endpoint returns 200 (idempotent success) even when the owner directory cannot be resolved, but it does **not** call `remove_user_push_subscription` in that path — so the in-memory subscription survives and the assertion fails.

## Reproduction

```
pytest tests/test_push_subscription_route.py::test_push_subscription_owner_validation
```

Fails with:
```
AssertionError: assert {'endpoint': 'https://ex', 'keys': {'auth': 'b', 'p256dh': 'a'}} is None
```

## Root cause

The `accounts_root` override is unrelated to push subscription deletion — subscriptions are stored in `_SUBSCRIPTIONS_STORAGE`, not in the accounts directory. The override was presumably added to test a fallback path but makes the delete a no-op while the assertion expects a real delete.

## Fix

Remove the `client.app.state.accounts_root = tmp_path / "does-not-exist"` line from the delete portion of the test. The "invalid accounts_root" fallback case is separately covered by `test_push_subscription_falls_back_to_default_dataset`.

## Acceptance criteria

- [ ] `test_push_subscription_owner_validation` passes
- [ ] Test still validates the full cycle: subscribe → verify → delete → verify deleted
- [ ] No production code change required — test correctness fix only

--- a/.github/workflows/backend-integration.yml
+++ b/.github/workflows/backend-integration.yml
@@ -6,7 +6,11 @@ on:
     types: [opened, reopened, synchronize]
   push:
     branches: [main]
-
+    
+permissions:
+  contents: read
+  id-token: write
+  
 jobs:
   integration-tests:
     runs-on: ubuntu-latest

--- a/.github/workflows/deploy-lambda.yml
+++ b/.github/workflows/deploy-lambda.yml
@@ -66,8 +66,8 @@ jobs:
             echo "AWS_REGION is not configured (set as secret or repository variable)" >&2
             missing=1
           fi
-          if [ -z "${{ secrets.AWS_ROLE_TO_ASSUME }}" ]; then
-            echo "AWS_ROLE_TO_ASSUME secret is not configured" >&2
+          if [ -z "${{ secrets.AWS_ROLE_TO_ASSUME }}${{ vars.AWS_ROLE_TO_ASSUME }}" ]; then
+            echo "AWS_ROLE_TO_ASSUME is not configured (set as secret or repository variable)" >&2
             missing=1
           fi
           if [ "$missing" -eq 1 ]; then exit 1; fi

--- a/frontend/src/AuthContext.tsx
+++ b/frontend/src/AuthContext.tsx
@@ -1,30 +1,14 @@
-import { createContext, useContext, useState, useCallback } from "react";
-import type { ReactNode } from "react";
-import {
-  loadStoredAuthUser,
-  persistStoredAuthUser,
-  type StoredUserProfile,
-} from "./authStorage";
+import { useState, useCallback, useContext } from 'react';
+import type { ReactNode } from 'react';
+import { loadStoredAuthUser, persistStoredAuthUser } from './authStorage';
+import { AuthContext } from './contexts/auth';
 
-export type UserProfile = StoredUserProfile;
-
-interface AuthContextValue {
-  user: UserProfile | null;
-  setUser: (u: UserProfile | null) => void;
-}
-
-// Default context used when no provider is present. The setter is a no-op so
-// components can still call it safely in tests or non-authenticated scenarios.
-export const AuthContext = createContext<AuthContextValue>({
-  user: null,
-  setUser: () => {},
-});
+export type { UserProfile } from './contexts/auth';
+export { AuthContext } from './contexts/auth';
 
 export function AuthProvider({ children }: { children: ReactNode }) {
-  const [user, setUserState] = useState<UserProfile | null>(() =>
-    loadStoredAuthUser(),
-  );
-  const setUser = useCallback((u: UserProfile | null) => {
+  const [user, setUserState] = useState(() => loadStoredAuthUser());
+  const setUser = useCallback((u: Parameters<typeof setUserState>[0]) => {
     setUserState(u);
     persistStoredAuthUser(u);
   }, []);
@@ -38,4 +22,3 @@ export function AuthProvider({ children }: { children: ReactNode }) {
 export function useAuth() {
   return useContext(AuthContext);
 }
-

--- a/frontend/src/PriceRefreshContext.tsx
+++ b/frontend/src/PriceRefreshContext.tsx
@@ -1,16 +1,8 @@
-import { createContext, useContext, useState, type ReactNode } from "react";
+import { useState, useContext, type ReactNode } from 'react';
+import { PriceRefreshContext } from './contexts/priceRefresh';
 
-interface PriceRefreshContextValue {
-  lastRefresh: string | null;
-  setLastRefresh: (ts: string | null) => void;
-}
-
-const defaultValue: PriceRefreshContextValue = {
-  lastRefresh: null,
-  setLastRefresh: () => {},
-};
-
-export const PriceRefreshContext = createContext<PriceRefreshContextValue>(defaultValue);
+export type { PriceRefreshContextValue } from './contexts/priceRefresh';
+export { PriceRefreshContext } from './contexts/priceRefresh';
 
 export function PriceRefreshProvider({ children }: { children: ReactNode }) {
   const [lastRefresh, setLastRefresh] = useState<string | null>(null);

--- a/frontend/src/RouteContext.tsx
+++ b/frontend/src/RouteContext.tsx
@@ -1,6 +1,6 @@
-import { createContext, useContext, type ReactNode } from "react";
-import { useRouteMode } from "./hooks/useRouteMode";
-import type { Mode } from "./modes";
+import { createContext, type ReactNode } from 'react';
+import { useRouteMode } from './hooks/useRouteMode';
+import type { Mode } from './modes';
 
 interface RouteContextValue {
   mode: Mode;
@@ -11,17 +11,12 @@ interface RouteContextValue {
   setSelectedGroup: (s: string) => void;
 }
 
-const RouteContext = createContext<RouteContextValue | undefined>(undefined);
+export const RouteContext = createContext<RouteContextValue | undefined>(undefined);
 
 export function RouteProvider({ children }: { children: ReactNode }) {
   const value = useRouteMode();
   return <RouteContext.Provider value={value}>{children}</RouteContext.Provider>;
 }
 
-// eslint-disable-next-line react-refresh/only-export-components
-export function useRoute() {
-  const ctx = useContext(RouteContext);
-  if (!ctx) throw new Error("useRoute must be used within RouteProvider");
-  return ctx;
-}
-
+// Re-export hook for backward compatibility.
+export { useRoute } from './hooks/useRoute';

--- a/frontend/src/RouteContext.tsx
+++ b/frontend/src/RouteContext.tsx
@@ -1,22 +1,13 @@
-import { createContext, type ReactNode } from 'react';
+import { type ReactNode } from 'react';
 import { useRouteMode } from './hooks/useRouteMode';
-import type { Mode } from './modes';
+import { RouteContext } from './contexts/route';
 
-interface RouteContextValue {
-  mode: Mode;
-  setMode: (m: Mode) => void;
-  selectedOwner: string;
-  setSelectedOwner: (s: string) => void;
-  selectedGroup: string;
-  setSelectedGroup: (s: string) => void;
-}
-
-export const RouteContext = createContext<RouteContextValue | undefined>(undefined);
+export type { RouteContextValue } from './contexts/route';
+export { RouteContext } from './contexts/route';
 
 export function RouteProvider({ children }: { children: ReactNode }) {
   const value = useRouteMode();
   return <RouteContext.Provider value={value}>{children}</RouteContext.Provider>;
 }
 
-// Re-export hook for backward compatibility.
 export { useRoute } from './hooks/useRoute';

--- a/frontend/src/contexts/auth.ts
+++ b/frontend/src/contexts/auth.ts
@@ -1,0 +1,16 @@
+import { createContext } from 'react';
+import type { StoredUserProfile } from '../authStorage';
+
+export type UserProfile = StoredUserProfile;
+
+export interface AuthContextValue {
+  user: UserProfile | null;
+  setUser: (u: UserProfile | null) => void;
+}
+
+// Default context used when no provider is present. The setter is a no-op so
+// components can still call it safely in tests or non-authenticated scenarios.
+export const AuthContext = createContext<AuthContextValue>({
+  user: null,
+  setUser: () => {},
+});

--- a/frontend/src/contexts/priceRefresh.ts
+++ b/frontend/src/contexts/priceRefresh.ts
@@ -1,0 +1,14 @@
+import { createContext } from 'react';
+
+export interface PriceRefreshContextValue {
+  lastRefresh: string | null;
+  setLastRefresh: (ts: string | null) => void;
+}
+
+const defaultValue: PriceRefreshContextValue = {
+  lastRefresh: null,
+  setLastRefresh: () => {},
+};
+
+export const PriceRefreshContext =
+  createContext<PriceRefreshContextValue>(defaultValue);

--- a/frontend/src/contexts/route.ts
+++ b/frontend/src/contexts/route.ts
@@ -1,0 +1,13 @@
+import { createContext } from 'react';
+import type { Mode } from '../modes';
+
+export interface RouteContextValue {
+  mode: Mode;
+  setMode: (m: Mode) => void;
+  selectedOwner: string;
+  setSelectedOwner: (s: string) => void;
+  selectedGroup: string;
+  setSelectedGroup: (s: string) => void;
+}
+
+export const RouteContext = createContext<RouteContextValue | undefined>(undefined);

--- a/frontend/src/hooks/useRoute.ts
+++ b/frontend/src/hooks/useRoute.ts
@@ -1,5 +1,5 @@
 import { useContext } from 'react';
-import { RouteContext } from '../RouteContext';
+import { RouteContext } from '../contexts/route';
 
 export function useRoute() {
   const ctx = useContext(RouteContext);

--- a/frontend/src/hooks/useRoute.ts
+++ b/frontend/src/hooks/useRoute.ts
@@ -1,0 +1,8 @@
+import { useContext } from 'react';
+import { RouteContext } from '../RouteContext';
+
+export function useRoute() {
+  const ctx = useContext(RouteContext);
+  if (!ctx) throw new Error('useRoute must be used within RouteProvider');
+  return ctx;
+}

--- a/frontend/tests/unit/familyMvpRedirect.test.ts
+++ b/frontend/tests/unit/familyMvpRedirect.test.ts
@@ -41,9 +41,6 @@ describe('getFamilyMvpRedirectPath', () => {
   });
 
   it('redirects non-MVP routes to transactions', () => {
-    // entryPath defaults to /portfolio but at runtime familyMvpEntryPath
-    // resolves to /transactions (first enabled FAMILY_MVP_ENTRY_CANDIDATE).
-    // These tests pass the default explicitly to confirm function behaviour.
     expect(getFamilyMvpRedirectPath('/market', '', true)).toBe('/transactions');
     expect(getFamilyMvpRedirectPath('/support', '', true)).toBe('/transactions');
     expect(getFamilyMvpRedirectPath('/performance/alex', '', true)).toBe('/transactions');
@@ -76,35 +73,6 @@ describe('getFamilyMvpRedirectPath', () => {
   it('supports overriding the entry path when a different MVP route is enabled', () => {
     expect(getFamilyMvpRedirectPath('/', '', true, '/performance')).toBe('/performance');
     expect(getFamilyMvpRedirectPath('/support', '', true, '/transactions')).toBe('/transactions');
-  it('redirects non-MVP routes to the configured entry path', () => {
-    expect(getFamilyMvpRedirectPath('/market', '')).toBe('/input');
-    expect(getFamilyMvpRedirectPath('/support', '')).toBe('/input');
-  });
-
-  it('does not redirect MVP routes', () => {
-    expect(getFamilyMvpRedirectPath('/portfolio', '', '/portfolio')).toBeNull();
-    expect(getFamilyMvpRedirectPath('/input', '')).toBeNull();
-    expect(getFamilyMvpRedirectPath('/transactions', '')).toBeNull();
-    expect(getFamilyMvpRedirectPath('/portfolio/alex', '')).toBeNull();
-    expect(getFamilyMvpRedirectPath('/performance/alex', '')).toBeNull();
-    expect(
-      getFamilyMvpRedirectPath('/performance/alex', '?range=1y')
-    ).toBeNull();
-  });
-
-  it('redirects bare root to default portfolio entry path', () => {
-    expect(getFamilyMvpRedirectPath('/', '')).toBe('/input');
-  });
-
-  it('redirects group query routes because group mode is non-MVP', () => {
-    expect(getFamilyMvpRedirectPath('/', '?group=kids')).toBe('/input');
-  });
-
-  it('supports overriding the entry path when a different MVP route is enabled', () => {
-    expect(getFamilyMvpRedirectPath('/', '', '/performance')).toBe(
-      '/performance'
-    );
-    expect(getFamilyMvpRedirectPath('/support', '', '/input')).toBe('/input');
   });
 
   it('skips redirecting when every family MVP route is disabled (null entryPath)', () => {

--- a/tests/test_push_subscription_route.py
+++ b/tests/test_push_subscription_route.py
@@ -1,3 +1,4 @@
+import os
 from pathlib import Path
 
 import pytest
@@ -12,12 +13,22 @@ from backend.common.storage import get_storage
 
 @pytest.fixture
 def client(tmp_path, monkeypatch):
+    # Redirect subscription storage to a fresh tmp file.
     monkeypatch.setattr(
         alert_utils,
         "_SUBSCRIPTIONS_STORAGE",
         get_storage(f"file://{tmp_path / 'push.json'}"),
     )
+    # Clear the in-memory cache so tests start from a known-empty state.
     alert_utils._PUSH_SUBSCRIPTIONS.clear()
+    # Unset DATA_BUCKET so _save_subscriptions/_load_subscriptions use the
+    # monkeypatched file storage rather than attempting (and silently failing)
+    # S3 calls.  When DATA_BUCKET is set, _save_subscriptions returns after the
+    # S3 path without ever writing to _SUBSCRIPTIONS_STORAGE, so the tmp file
+    # is never updated and a subsequent reload repopulates _PUSH_SUBSCRIPTIONS
+    # from the stale file written during the POST.
+    monkeypatch.delenv("DATA_BUCKET", raising=False)
+
     original_arn = alert_utils.config.sns_topic_arn
     alert_utils.config.sns_topic_arn = None
     original_disable_auth = backend_config.config.disable_auth
@@ -41,10 +52,6 @@ def test_push_subscription_owner_validation(client, tmp_path):
     resp_bad = client.post("/alerts/push-subscription/unknown", json=payload)
     assert resp_bad.status_code == 404
 
-    # Delete the subscription — do not override accounts_root here; that would
-    # cause the DELETE route to skip removal (subscriptions storage is separate
-    # from the accounts directory). accounts_root fallback behaviour is tested
-    # separately in test_push_subscription_falls_back_to_default_dataset.
     resp_del = client.delete(f"/alerts/push-subscription/{owner}")
     assert resp_del.status_code == 200
     assert alert_utils.get_user_push_subscription(owner) is None

--- a/tests/test_push_subscription_route.py
+++ b/tests/test_push_subscription_route.py
@@ -41,7 +41,10 @@ def test_push_subscription_owner_validation(client, tmp_path):
     resp_bad = client.post("/alerts/push-subscription/unknown", json=payload)
     assert resp_bad.status_code == 404
 
-    client.app.state.accounts_root = tmp_path / "does-not-exist"
+    # Delete the subscription — do not override accounts_root here; that would
+    # cause the DELETE route to skip removal (subscriptions storage is separate
+    # from the accounts directory). accounts_root fallback behaviour is tested
+    # separately in test_push_subscription_falls_back_to_default_dataset.
     resp_del = client.delete(f"/alerts/push-subscription/{owner}")
     assert resp_del.status_code == 200
     assert alert_utils.get_user_push_subscription(owner) is None

